File: Media Ministry Manager/Custom Objects/Dialogs/Emails/ReceiptTypeDialog.vb
Public Class ReceiptTypeDialog
	Public Enum ReceiptType
		Tithes
		Love
		Offering
		Other
	End Enum

	Public ReadOnly Property Amount As Double
		Get
			Return nud_Amount.Value
		End Get
	End Property

	Public Property ReceiptOption As ReceiptType

	Public ReadOnly Property OtherOption As String
		Get
			Return txt_Other.Text
		End Get
	End Property

	Private Sub Btn_Ok_Click(sender As Object, e As EventArgs) Handles btn_Ok.Click
		If rdo_Other.Checked AndAlso String.IsNullOrWhiteSpace(txt_Other.Text) Then
			MessageBox.Show("You must enter something that says what this receipt is for. Please enter a value or select a preset type.", "Select Type", MessageBoxButtons.OK, MessageBoxIcon.Error)
			Return
		End If

		Me.DialogResult = DialogResult.OK
		Me.Close()
	End Sub

	Private Sub Btn_Cancel_Click(sender As Object, e As EventArgs) Handles btn_Cancel.Click
		Me.DialogResult = DialogResult.Cancel
		Me.Close()
	End Sub

	Private Sub Rdo_Other_CheckedChanged(sender As Object, e As EventArgs) Handles rdo_Other.CheckedChanged
		If rdo_Other.Checked Then
			Me.Size = New Size(440, 419)
		Else
			Me.Size = New Size(440, 368)
		End If

		ReceiptOption = ReceiptType.OTHER
		lbl_Other.Visible = rdo_Other.Checked
		txt_Other.Visible = rdo_Other.Checked
	End Sub

	Private Sub TithesCheckedChanged(sender As Object, e As EventArgs) Handles rdo_Tithes.CheckedChanged
		ReceiptOption = ReceiptType.TITHES
	End Sub

	Private Sub LoveOfferingCheckedChanged(sender As Object, e As EventArgs) Handles rdo_LoveOffering.CheckedChanged
		ReceiptOption = ReceiptType.LOVE
	End Sub

	Private Sub OfferingCheckedChanged(sender As Object, e As EventArgs) Handles rdo_Offering.CheckedChanged
		ReceiptOption = ReceiptType.OFFERING
	End Sub
End Class

File: Media Ministry Manager/Custom Objects/Dialogs/Emails/ReceiptTypeDialog.cs
using System;
using System.Drawing;
using System.Windows.Forms;

namespace MediaMinistry
{
    public partial class ReceiptTypeDialog
    {
        public enum ReceiptType
        {
            Tithes,
            Love,
            Offering,
            Other
        }

        public double Amount
        {
            get
            {
                return (double)nud_Amount.Value;
            }
        }

        public ReceiptType ReceiptOption { get; set; }

        public string OtherOption
        {
            get
            {
                return txt_Other.Text;
            }
        }

        public ReceiptTypeDialog()
        {
            InitializeComponent();
        }

        private void Btn_Ok_Click(object sender, EventArgs e)
        {
            if (rdo_Other.Checked && string.IsNullOrWhiteSpace(txt_Other.Text))
            {
                MessageBox.Show("You must enter something that says what this receipt is for. Please enter a value or select a preset type.", "Select Type", MessageBoxButtons.OK, MessageBoxIcon.Error);
                return;
            }

            DialogResult = DialogResult.OK;
            Close();
        }

        private void Btn_Cancel_Click(object sender, EventArgs e)
        {
            DialogResult = DialogResult.Cancel;
            Close();
        }

        private void Rdo_Other_CheckedChanged(object sender, EventArgs e)
        {
            if (rdo_Other.Checked)
            {
                Size = new Size(440, 419);
            }
            else
            {
                Size = new Size(440, 368);
            }

            ReceiptOption = ReceiptType.Other;
            lbl_Other.Visible = rdo_Other.Checked;
            txt_Other.Visible = rdo_Other.Checked;
        }

        private void TithesCheckedChanged(object sender, EventArgs e)
        {
            ReceiptOption = ReceiptType.Tithes;
        }

        private void LoveOfferingCheckedChanged(object sender, EventArgs e)
        {
            ReceiptOption = ReceiptType.Love;
        }

        private void OfferingCheckedChanged(object sender, EventArgs e)
        {
            ReceiptOption = ReceiptType.Offering;
        }
    }
}